src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -822,10 +821,10 @@ private Compilation CreateEmptyCompilation()
             }
 
             private async Task<(Compilation compilationWithGeneratedFiles, CompilationTrackerGeneratorInfo generatorInfo)> AddExistingOrComputeNewGeneratorInfoAsync(
+                SolutionState solution,
                 Compilation compilationWithoutGeneratedFiles,
                 CompilationTrackerGeneratorInfo generatorInfo,
                 Compilation? compilationWithStaleGeneratedTrees,
-                ISourceGeneratorTelemetryCollectorWorkspaceService? telemetryCollector,
                 CancellationToken cancellationToken)
```

Comment:
i'd like to move this funciton to the new file as well. i'll do so at the end after the PR is done.